### PR TITLE
Organize theme files

### PR DESCRIPTION
**Pull request by jessabean**
### Description
[RFC](https://docs.google.com/document/d/1zyHGfboYyhncV0MXnoCNipn1xWoD8IfAS_oVK5-FExk/edit#heading=h.rz1wnztilpui)

Proposed organization of theme files. Breaking up main theme file into more atomic pieces that include foundation styles and component styles, along with their variants.

### To Test
- This kind of touches every part of the app (ugh, sorry), so it will require looking around at all the pages
- Everything should look mostly the same and nothing obviously broken

### CODE REVIEW

File: services/ui-src/src/styles/components/link.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { ComponentStyleConfig } from "@chakra-ui/react";
+
+const baseStyles = {
+  textDecoration: "underline",
+  transition: "all 0.3s ease",
+};
+
+const primaryVarian = {
```
Comment (gmrabian) [inline services/ui-src/src/styles/components/link.ts:8]: typo
```suggestion
const primaryVariant = {
```

**Review by gmrabian — COMMENTED** [review]
I know this is still a draft but I think it's really cool

**Review by britt-mo — COMMENTED** [review]
disabled edit button styling 
Screenshot 2024-08-27 at 1 40 42 PM


Other than that, there are just smaller things that differ from val--buttons are a bit wider, `x Close` button in modals is positioned lower, card header text in this PR is bigger

**Comment by gmrabian** [comment]
> disabled edit button styling Screenshot 2024-08-27 at 1 40 42 PM
> 
> Other than that, there are just smaller things that differ from val--buttons are a bit wider, `x Close` button in modals is positioned lower, card header text in this PR is bigger

+1

Also links remain in the unclicked styling on both this and dev so we are keeping functionality, but is that intentional?

**Comment by jessabean** [comment]
> > disabled edit button styling Screenshot 2024-08-27 at 1 40 42 PM
> > Other than that, there are just smaller things that differ from val--buttons are a bit wider, `x Close` button in modals is positioned lower, card header text in this PR is bigger
> 
> +1
> 
> Also links remain in the unclicked styling on both this and dev so we are keeping functionality, but is that intentional?

Garrett, I'm not sure I understand this question! Can you show me an example? It's possibly a regression, but I think I need an example to be sure.

**Review by karla-vm — APPROVED** [review]

**Pull request merged**